### DISCUSSION TITLE
Heroku Postgres SSL issue - "Error: self signed certificate"

### DISCUSSION THREAD

**Discussion by saltedw** [alexandrainst/node-red-contrib-postgresql#47] [Q&A]
I'm attempting to connect to a Heroku PostgreSQL hosted database. They require SSL but use a self-signed cert that node seems not to like. Connections give the following error: "Error: self signed certificate"

Googling shows most recommend setting an SSL value of rejectUnauthorized: false (below).

Now I see there is a way to pass in fully custom settings via msg.pgConfig, but i would love if this rejectUnauthorized was built in so i don't have to pass my credentials and full connection in via pgConfig. Is there another way to configure rejectUnauthorized? Can i just pass in ssl: { rejectUnauthorized: false } for example, and let it use the rest via the node-red UI for username, connection, etc?

Thanks much.

*reference materials: https://stackoverflow.com/questions/61097695/self-signed-certificate-error-during-query-the-heroku-hosted-postgres-database 

I was able to get my connection working to heroku postgres by passing the following into msg.pgConfig
{   "user": "blablabla",    "password": "blablabla",    "host": "blablabla.compute-1.amazonaws.com",   "database": "blablabla",    "port": "5432",   "ssl": {              "required": true,             "rejectUnauthorized": false           },   "connectionTimeoutMillis": "10000",   "idle_in_transaction_session_timeout": "1000"  }

**Comment by Alkarex** [comment]
Hello,
This node is a wrapper around node-postgres, which seem to have the needed features
https://node-postgres.com/features/ssl#self-signed-cert

I cannot do it at the moment, but it looks relatively easy to implement. PR welcome
